src/lib.rs

```diff
@@ -144,8 +140,10 @@ pub async fn start_lemmy_server() -> Result<(), LemmyError> {
     });
   }
 
-  #[cfg(feature = "prometheus-metrics")]
-  serve_prometheus(settings.prometheus.as_ref(), context.clone());
+  match settings.prometheus.is_some() {
+    true => serve_prometheus_metrics(settings.prometheus.as_ref(), context.clone()),
+    false => info!("No Prometheus config provided, will not start Prometheus server"),
+  }
```

Comment:
Use `if let Some(config) = settings.prometheus`, then you dont need any panic. Also I wouldnt log anything if prometheus config is unset.